compiler/optimizer/LoopReducer.cpp

```diff
@@ -1088,23 +1088,24 @@ TR_LoopReducer::generateArraycopy(TR_InductionVariable * indVar, TR::Block * loo
    if (arraycopyLoop.getStoreAddress()->getIncrement() < 0)
       //&& (storeAddr->getFirstChild()->getSymbol()->getRegisterMappedSymbol() == loadAddr->getFirstChild()->getSymbol()->getRegisterMappedSymbol()))
       {
-      if (comp()->cg()->getSupportsPrimitiveArrayCopy())
-         arraycopy->setBackwardArrayCopy(true); /* bit available only to primitive arraycopy */
```

Comment:
Yes but the only call to this method is guarded by `getSupportsPostProcessArrayCopy` which only z Systems implements returning `true`. It is no surprise that z Systems is (currently) the only platform that answers `true` to `getSupportsPrimitiveArrayCopy` so the backwards array copy check there does not matter.